src/components/PercentAlertForm/PercentAlertForm.tsx

```
@@ -7,147 +7,284 @@ import { SelectField } from "@/components/ui/select-field";
+
+  const { coins, exchanges, isLoading: isLoadingData, error: dataError } = useCoinAndExchangeData();
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
   const [error, setError] = useState<string | null>(null);
 
+  React.useEffect(() => {
+    if (coins.length > 0 && form.getValues("coin") === "") {
+      form.setValue("coin", coins[0]);
+    }
+  }, [coins, form]);
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)